src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -589,12 +590,12 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
       BuildOptions defaultBuildOptions)
       throws DependencyEvaluationException, ConfiguredTargetFunctionException,
           AspectCreationException, InterruptedException {
-    // Create the map from attributes to set of (target, configuration) pairs.
-    OrderedSetMultimap<DependencyKind, Dependency> depValueNames;
```

Comment:
I've lost track of what kind of "dependency" descriptor can get created and passed through what parts of the logic.

Originally I thought depValueNames. could get assigned to both configuration transition dependencies and direct ones such as HOST and NULL. is that no longer the case?

To be clear, that's all the more reason for this work you're doing. My main interest in the final outcome is really getting naming that clarifies the current reality as clearly as possible, so I'm excited about this.

Reply:
I also thought that was the case, but it turns out that it's not true. Every time the Dependency type was used before this change, it was either a ConfigurationTransitionDependency, or one of the NullConfigurationDependency or ExplicitConfigurationDependency. This is why the hasExplicitConfiguration() method is entirely removed: it's not needed.

One of the major motivators of this change is clarifying what data exists at what points during dependency resolution.

Reply:
Wow, you're right! Can't believe how much this logic has moved past me over time.